R/render.R

```diff
@@ -333,6 +337,76 @@ render <- function(input,
     }, add = TRUE)
     env$metadata <- yaml_front_matter
 
+    # hook up output_source to evaluate if specified
```

Comment:
Yes, exactly. It would all go in the pre_knit hook. The output_format
object would also gain another member (an on_exit hook). The html_notebook
format would implement on_exit to reverse out it's various global state
changes.

On Sat, May 7, 2016 at 1:20 PM, Kevin Ushey <notifications@github.com>
wrote:

> In R/render.R
> <https://github.com/rstudio/rmarkdown/pull/693#discussion_r62419710>:
>
> > @@ -333,6 +337,76 @@ render <- function(input,
> >      }, add = TRUE)
> >      env$metadata <- yaml_front_matter
> >
> > +    # hook up output_source to evaluate if specified
>
> I see -- the idea then would be that the format can do any of the work it
> needs to in the pre_knit hook; does that sound correct? (Ie, we can move
> all of the code I added to render into the format's pre_knit hook). I
> think that should be doable.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/rstudio/rmarkdown/pull/693/files/86ff097e5718178104755c6bba70811faa193802#r62419710>
>